### PR TITLE
TMA-354 Update the obfuscation-service.ts to handle obfuscating objec…

### PR DESCRIPTION
**Pull request by jashelb90**
…ts/arrays. This is to ensure all field names are retained and the structure is retained.

### CODE REVIEW

**Review by des-hippo — DISMISSED** [review]
All looks good. And nice use of recursion.

File: event-processing/obfuscation/tests/unit/test-helper.ts [hunk 1]
```diff
@@ -40,9 +40,9 @@ export class TestHelper {
         event_name: 'AUTHENTICATION_ATTEMPT',
         component_id: 'AUTH',
         user: {
-            transaction_id: ObfuscationService.obfuscate('a52f6f87', 'secret-1-value'),
-            email: ObfuscationService.obfuscate('foo@bar.com', 'secret-1-value'),
-            phone: ObfuscationService.obfuscate('07711223344', 'secret-1-value'),
+            transaction_id: ObfuscationService.obfuscateField('a52f6f87', 'secret-1-value'),
+            email: ObfuscationService.obfuscateField('foo@bar.com', 'secret-1-value'),
+            phone: ObfuscationService.obfuscateField('07711223344', 'secret-1-value'),
```
Comment (phantas) [inline event-processing/obfuscation/tests/unit/test-helper.ts:45]: can you please also add an unexpected field, say "foo", to check it is obfuscated as well

**Review by phantas — CHANGES_REQUESTED** [review]

File: event-processing/obfuscation/tests/unit/app.test.ts [hunk 2]
```diff
@@ -207,7 +284,7 @@ describe('Unit test for app handler', function () {
         const key : string = "My secret key";
         const expectedResult = "5d224b25388f72dc5e329dd68385680535f6d9bda65c5f830631d72e255b9f95"
 
-        const result: string = ObfuscationService.obfuscate(message, key);
+        const result: string = ObfuscationService.obfuscateField(message, key);
```
[the remark this answers is not in the document]
Comment (jashelb90) [reply event-processing/obfuscation/tests/unit/app.test.ts:287]: That is filled in now as a tech debt ticket.

File: event-processing/obfuscation/tests/unit/test-helper.ts (hunk 1 above)
Comment (jashelb90) [reply event-processing/obfuscation/tests/unit/test-helper.ts:45]: @phantas do you agree with this or would you prefer adding an unexpected field for posterity?

Comment (phantas) [reply event-processing/obfuscation/tests/unit/test-helper.ts:45]: I am ok with that

**Review by phantas — APPROVED** [review]

**Pull request merged**